test/parallel/xyk-pallet.API.pool.promoting.test.ts

```diff
@@ -0,0 +1,198 @@
+  );
+});
+
+test("GIVEN a promoted pool WHEN more pools gets activated THEN shares are decreased and equally divided among all the activated pools", async () => {
+  await activateLiquidity(
+    testUser1.keyRingPair,
+    liqIdPool1,
+    Assets.DEFAULT_AMOUNT.divn(2)
+  );
+  await waitForRewards(testUser1, liqIdPool1);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(liqIdPool2.toNumber(), 20),
+    Assets.promotePool(liqIdPool3.toNumber(), 20),
+    Sudo.sudoAs(
+      testUser2,
+      Xyk.activateLiquidity(liqIdPool2, Assets.DEFAULT_AMOUNT.divn(2))
+    ),
+    Sudo.sudoAs(
+      testUser3,
+      Xyk.activateLiquidity(liqIdPool3, Assets.DEFAULT_AMOUNT.divn(2))
+    ),
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqIdPool1))
+  );
+  const rewardsInfoOnlyPool1 = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIdPool1
+  );
+
+  await waitForRewards(testUser2, liqIdPool2);
+  await waitForRewards(testUser3, liqIdPool3);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqIdPool1)),
+    Sudo.sudoAs(testUser2, Xyk.claimRewardsAll(liqIdPool2)),
+    Sudo.sudoAs(testUser3, Xyk.claimRewardsAll(liqIdPool3))
+  );
+
+  const rewardsInfoPool1Before = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIdPool1
+  );
+  const rewardsInfoPool2Before = await getRewardsInfo(
+    testUser2.keyRingPair.address,
+    liqIdPool2
+  );
+  const rewardsInfoPool3Before = await getRewardsInfo(
+    testUser3.keyRingPair.address,
+    liqIdPool3
+  );
+
+  await waitForRewards(testUser1, liqIdPool1);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqIdPool1)),
+    Sudo.sudoAs(testUser2, Xyk.claimRewardsAll(liqIdPool2)),
+    Sudo.sudoAs(testUser3, Xyk.claimRewardsAll(liqIdPool3))
+  );
+
+  const rewardsInfoPool1After = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqIdPool1
+  );
+  const rewardsInfoPool2After = await getRewardsInfo(
+    testUser2.keyRingPair.address,
+    liqIdPool2
+  );
+  const rewardsInfoPool3After = await getRewardsInfo(
+    testUser3.keyRingPair.address,
+    liqIdPool3
+  );
+  const differenceRewardPool1 = rewardsInfoPool1After.rewardsAlreadyClaimed
+    .sub(rewardsInfoPool1Before.rewardsAlreadyClaimed)
+    .sub(rewardsInfoOnlyPool1.rewardsAlreadyClaimed);
+
+  const differenceRewardPool2 = rewardsInfoPool2After.rewardsAlreadyClaimed.sub(
+    rewardsInfoPool2Before.rewardsAlreadyClaimed
+  );
+
+  const differenceRewardPool3 = rewardsInfoPool3After.rewardsAlreadyClaimed.sub(
+    rewardsInfoPool3Before.rewardsAlreadyClaimed
+  );
+
+  const ratioRewardPools = differenceRewardPool3
+    .mul(new BN(100))
+    .div(differenceRewardPool1);
+
+  expect(differenceRewardPool2).bnEqual(differenceRewardPool3);
+  //difference between the supposed rewards Pool1 and rewards of the next pools in percent
+  expect(ratioRewardPools.sub(new BN(100))).bnLt(new BN(11));
```

Comment:
@AlexChetverov  the setup looks awesome, but this may fail when running all tests together, @goncer  think on some deterministic way to test this.